### DISCUSSION TITLE
Looking for a way to exclude local fields from being sent to server

### DISCUSSION THREAD

**Discussion by robertherber** [urql-graphql/urql#2590] [Q&A] [answered]
I'm using a local resolver for a `isAvailableOffline` field. I'd like a way to specify this field in the client in my schema (extending my `Task` type) but prevent it from being sent to the server.

My current workaround is to add the `isAvailableOffline` that always resolves to false on the backend. But I'd rather keep local state on the client only.

Is there a way to do this currently? In Apollo they have a `@client` directive for this - maybe URQL could introduce something similar?

**Comment by JoviDeCroock** [comment, marked as answer]
You can write an exchange that does just that, we haven't implemented local state as we haven't seen many use cases that demandes local extension of a server schema.

In this case it can be a pretty [basic exchange](https://formidable.com/open-source/urql/docs/advanced/authoring-exchanges/#the-rules-of-exchanges) that we put before the `fetchExchange` that removes the local fields before forwarding to the fetch one and adds them back when returning
